solidity/contracts/converter/BancorConverter.sol

```diff
@@ -779,7 +778,50 @@ contract BancorConverter is IBancorConverter, SmartTokenController, Managed, Con
         }
 
         // execute the conversion and pass on the ETH with the call
-        return bancorNetwork.convertForPrioritized2.value(msg.value)(_path, _amount, _minReturn, msg.sender, _block, _v, _r, _s);
+        return bancorNetwork.convertForPrioritized3.value(msg.value)(_path, _amount, _minReturn, msg.sender, _amount, _block, _v, _r, _s);
+    }
+
+    /**
+        @dev allows a user to convert BNT that was sent from another blockchain into any other
+        token on the BancorNetwork without specifying the amount of BNT to be converted, but
+        rather by providing the xTransferId which allows us to get the amount from BancorX.
+
+        @param _path             conversion path, see conversion path format in the BancorNetwork contract
+        @param _minReturn        if the conversion results in an amount smaller than the minimum return - it is cancelled, must be nonzero
+        @param _conversionId     pre-determined unique (if non zero) id which refers to this transaction 
+        @param _block            if the current block exceeded the given parameter - it is cancelled
+        @param _v                (signature[128:130]) associated with the signer address and helps to validate if the signature is legit
+        @param _r                (signature[0:64]) associated with the signer address and helps to validate if the signature is legit
+        @param _s                (signature[64:128]) associated with the signer address and helps to validate if the signature is legit
+
+        @return tokens issued in return
+    */
+    function completeXConversion(
+        IERC20Token[] _path,
+        uint256 _minReturn,
+        uint256 _conversionId,
+        uint256 _block,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    )
+        public
+        returns (uint256)
+    {
+        IBancorX bancorX = IBancorX(registry.addressOf(ContractIds.BANCOR_X));
+        IBancorNetwork bancorNetwork = IBancorNetwork(registry.addressOf(ContractIds.BANCOR_NETWORK));
+
+        // verify that the first token in the path is BNT
+        require(_path[0] == address(token));
```

Comment:
need to check it against BNT token, not against the converter token (the current code will also allow calling it for other converters, assuming the path is correct)